Marketplace/MarketplacePackager.ps1

```diff
@@ -0,0 +1,171 @@
+#endregion resources
+
+#region CreateUIDefinition
+   $partName=New-Object System.Uri("/Artifacts/CreateUiDefinition.json", [System.UriKind]"Relative")
+   $part=$ZipPackage.CreatePart($partName, "text/json")
+   $relationship=$ZipPackage.CreateRelationShip($partName, [System.IO.Packaging.TargetMode]::Internal, "http://schemas.microsoft.com/azpkg/2013/12/artifact")
+   $bytes=[System.IO.File]::ReadAllBytes("$source\Artifacts\CreateUiDefinition.json")
+   $stream=$part.GetStream()
+   $stream.Write($bytes, 0, $bytes.Length)
+   $stream.Close()
+
+   $guid=[guid]::NewGuid()
+   $guid=$guid.ToString("N")
+   $uri=New-Object System.Uri("/$guid/Artifact", [System.UriKind]"Relative")
+   $relationship=$part.CreateRelationShip($uri , [System.IO.Packaging.TargetMode]::Internal, "http://schemas.microsoft.com/azpkg/2013/12/deploymenttemplateproperties")
+   $part=$ZipPackage.CreatePart($uri, "text/xml")
+   $artifactContent = @"
+<?xml version="1.0"?>
+<Artifact xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <Name>createuidefinition</Name>
+  <Type>Custom</Type>
+  <Path>Artifacts\CreateUiDefinition.json</Path>
+  <IsDefault>false</IsDefault>
+</Artifact>
+"@
+   $enc = [system.Text.Encoding]::UTF8
+   $bytes=$enc.GetBytes($artifactContent)
+   $stream=$part.GetStream()
+   $stream.Write($bytes, 0, $bytes.Length)
+   $stream.Close()
+#endregion CreateUIDefinition
+
+#region ARM Template
+   $partName=New-Object System.Uri("/Artifacts/mainTemplate.json", [System.UriKind]"Relative")
+   $part=$ZipPackage.CreatePart($partName, "text/json")
+   $relationship=$ZipPackage.CreateRelationShip($partName, [System.IO.Packaging.TargetMode]::Internal, "http://schemas.microsoft.com/azpkg/2013/12/artifact")
+   $bytes=[System.IO.File]::ReadAllBytes("$source\Artifacts\mainTemplate.json")
+   $stream=$part.GetStream()
+   $stream.Write($bytes, 0, $bytes.Length)
+   $stream.Close()
+   
+   $guid=[guid]::NewGuid()
+   $guid=$guid.ToString("N")
+   $uri=New-Object System.Uri("/$guid/Artifact", [System.UriKind]"Relative")
+   $relationship=$part.CreateRelationShip($uri , [System.IO.Packaging.TargetMode]::Internal, "http://schemas.microsoft.com/azpkg/2013/12/deploymenttemplateproperties")
+   $part=$ZipPackage.CreatePart($uri, "text/xml")
+
+   $artifactContent = @"
+<?xml version="1.0"?>
+<Artifact xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
+  <Name>DefaultTemplate</Name>
+  <Type>Template</Type>
+  <Path>Artifacts\mainTemplate.json</Path>
+  <IsDefault>true</IsDefault>
+</Artifact>
+"@
+
+   $enc = [system.Text.Encoding]::UTF8
+   $bytes=$enc.GetBytes($artifactContent)
+   $stream=$part.GetStream()
+   $stream.Write($bytes, 0, $bytes.Length)
+   $stream.Close()
+
+   Write-Host "Azpkg is saved into '$zipArchive'" -ForegroundColor Green
+#endregion ARM Template
+} catch {
+   Write-Host "Azpkg wasn't created. $_" -ForegroundColor Magenta
+}
+
+$ZipPackage.Close()
```

Comment:
You might want to move this close into the Catch block too, or add a finally block with the close statement.